src/pubsub/src/subscriber/leaser.rs

```diff
@@ -39,4 +103,82 @@ pub(crate) mod tests {
             async fn extend(&self, ack_ids: Vec<String>);
         }
     }
+
+    #[tokio::test]
+    async fn ack() {
+        let mut mock = MockStub::new();
+        mock.expect_acknowledge().times(1).return_once(|r, o| {
+            assert_eq!(
+                r.subscription,
+                "projects/my-project/subscriptions/my-subscription"
+            );
+            assert_eq!(r.ack_ids, test_ids(0..10));
+            assert!(
+                format!("{o:?}").contains("NeverRetry"),
```

Comment:
Surprised there is not a more direct way to compare these. Maybe a comparison function could be useful for tests in the future (not necessary for this PR and might be overkill).